node/rest/src/lib.rs

```diff
@@ -95,9 +96,15 @@ impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
 }
 
 impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
-    fn spawn_server(&mut self, rest_ip: SocketAddr) {
+    fn spawn_server(&mut self, rest_ip: SocketAddr, allowed_origins: Vec<String>) {
+        let allowed_origins = if allowed_origins.is_empty() {
+            AllowOrigin::any()
+        } else {
+            let origins = allowed_origins.iter().map(|s| HeaderValue::from_str(s).expect("Invalid CORS origin"));
+            AllowOrigin::list(origins)
+        };
```

Comment:
IIRC this doesn't parse `*` correctly. From one of my own projects I remember having to check for `*` explicitly and return `AllowOrigin::any()` in that case. This is totally anecdotal though.

Reply:
Snippet from my code where I parse the origins from a comma-separated string and handle the wildcard case explicitly. Sadly I didn't put a comment there explaining the necessity of handling this case specifically, so I can't be 100% sure, but if I remember correctly the origins didn't work correctly when parsing `*` as a `HeaderValue`.

![CleanShot 2023-11-07 at 13 35 00@2x](https://github.com/AleoHQ/snarkOS/assets/4216049/8a4f5cde-81e2-4d7f-b987-bb8ed47adfde)

Reply:
in this case it is expected to leave the vector empty in order to accept `*`